src/app/mycart/my-cart.component.ts

```diff
@@ -1,15 +1,82 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router} from '@angular/router';
+import { ProductsService } from '../products/products.service';
 
 @Component({
   selector: 'app-my-cart',
   templateUrl: './my-cart.component.html',
   styleUrls: ['./my-cart.component.css']
 })
 export class MyCartComponent implements OnInit {
+  name:string;
+  productsList:any=[];
+  myCart:any=[];
+  cartValues=[];
+  totalAmount:number=0;
 
-  constructor() { }
+  constructor(private router:Router,
+              private route:ActivatedRoute,
+              private productsService:ProductsService)
+              { }
 
   ngOnInit(): void {
+    this.name=this.route.snapshot.paramMap.get('name');
+    this.productsService.getProducts().subscribe(item=>{
+      this.productsList=item;
+
+      this.cartValues=JSON.parse(localStorage.getItem('cartValue')) || [];
+      console.log("first",this.cartValues); 
+
+      for(let i=0;i<this.productsList.length;i++){
+        if(this.productsList[i].name===this.name && 
+          !this.cartValues.map(item=>item.name).includes(this.name)){
+            this.cartValues.push(this.productsList[i]);
+            console.log("after push",this.productsList[i]);
+            localStorage.setItem('cartValue',JSON.stringify(this.cartValues));
```

Comment:
you shouldn't really do this within every iteration. Expensive operation right?

Reply:
can i create a service and then use only cartvalues in mycart file?